cmd/snap/inhibit.go

```diff
@@ -73,47 +73,30 @@ var isGraphicalSession = func() bool {
 	return os.Getenv("DISPLAY") != "" || os.Getenv("WAYLAND_DISPLAY") != ""
 }
 
-var hasZenityExecutable = func() bool {
-	return osutil.ExecutableExists("zenity")
+var pendingRefreshNotification = func(refreshInfo *client.PendingSnapRefreshInfo) error {
+	userclient := client.New()
```

Comment:
Here, `userclient` will try to multiplex API calls to all available session agents on the system. In this case though, we probably want to only talk to the matching user's session agent.  Maybe we need a version of `New()` that lets us specify a subset of user IDs to contact?

While there is access control in the session agent to reject calls from other uids, I suspect we're not hitting these errors because other user's `$XDG_RUNTIME_DIR`s are not readable, so aren't picked up by the glob. It'd be better not to rely on this though.

Reply:
Great point, I did notice the multiplexing aspect of doMany in the client, agree it makes sense to make it userid-based.

Reply:
I've proposed https://github.com/snapcore/snapd/pull/11072 as a prerequisite for addressing this point.

Reply:
Done.